File: Models/Domain/Course.cs
namespace SCNProfessor.Models.Domain
{
    public class Course
    {
        private int id;
        private string name;
        private int cycle;
        private bool statusCourse;
        private string description;
        private int idProfessor;

        public Course()
        {
        }

        public Course(int id, string name, int cycle, bool statusCourse, string description, int idProfessor)
        {
            this.Id = id;
            this.Name = name;
            this.Cycle = cycle;
            this.StatusCourse = statusCourse;
            this.Description = description;
            this.IdProfessor = idProfessor;
        }

        public int Id { get => id; set => id = value; }
        public string Name { get => name; set => name = value; }
        public int Cycle { get => cycle; set => cycle = value; }
        public bool StatusCourse { get => statusCourse; set => statusCourse = value; }
        public string Description { get => description; set => description = value; }
        public int IdProfessor { get => idProfessor; set => idProfessor = value; }
    }
}

File: SCN-FrontEndProfessor/wwwroot/js/course.js
document.addEventListener('DOMContentLoaded', function () {
    const coursesList = document.getElementById('courses-list');

    // Función para cargar los cursos desde la API
    async function loadCourses() {
        try {
            const response = await fetch('https://localhost:44388/api/Course/GetAllCourses'); // Cambia la URL por la correcta de tu API
            if (!response.ok) {
                throw new Error('Error al cargar los cursos');
            }
            const courses = await response.json();

            // Limpiamos el contenedor antes de agregar los cursos
            coursesList.innerHTML = '';

            // Recorremos los cursos y los agregamos al contenedor
            courses.forEach(course => {
                const courseElement = document.createElement('div');
                courseElement.classList.add('col-md-4', 'course-card');

                // Se define el HTML para cada curso
                courseElement.innerHTML = `
                    <div class="card">
                        <div class="card-body">
                            <h5 class="card-title">${course.name}</h5>
                            <p class="card-text">${course.description}</p>
                            <p><strong>Cycle:</strong> ${course.cycle}</p>
                            <p><strong>Status:</strong> ${course.statusCourse}</p>
                        </div>
                    </div>
                `;

                // Añadimos el curso al contenedor de cursos
                coursesList.appendChild(courseElement);
            });
        } catch (error) {
            console.error(error);
            // Si hay un error, puedes mostrar un mensaje
            coursesList.innerHTML = '<p class="error-message">No se pudieron cargar los cursos. Intenta más tarde.</p>';
        }
    }

    // Llamamos a la función para cargar los cursos cuando la página haya cargado
    loadCourses();
});
